lib/BC/InstructionLifter.cpp

```diff
@@ -328,7 +328,8 @@ llvm::Value *InstructionLifter::LoadRegValue(llvm::BasicBlock *block,
                                              std::string_view reg_name) const {
   auto ptr = LoadRegAddress(block, state_ptr, reg_name);
   CHECK_NOTNULL(ptr);
-  auto ptr_ty = ptr->getType()->getPointerElementType();
+  // NOTE(alex): This isn't right. Not sure how to solve this right now.
+  auto ptr_ty = impl->word_type;
```

Comment:
Then change the API to fix this. Overall the `InstructionLifter` is totally evil and I regret it and so much more. We know that we can get the value type by `reg->type` if we can find `reg` via its name. However, we may also be dealing with "local var" regs, e.g. `BRANCH_TAKEN`. Those are `alloca`s for which the value type can be deduced. Finally, there is the case that t might find a global. I think instead of `auto ptr = LoadRegAddress(...)`, there should be something like `auto [ptr, val_type] = LoadRegAddress(...)`.

Reply:
[7fdcb51](https://github.com/lifting-bits/remill/pull/596/commits/7fdcb510f2c64d4aa70bc21df414538bb1b16159)